File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        numStr = str(s)
        num = int(numStr, 2)

        count = 0
        while num != 1:

            if (num % 2 == 0):
                num = num // 2
                count += 1
            
            else:
                num = num + 1
                count += 1
        
        return count
            


        

File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.cpp
class Solution {
public:
    int numSteps(string s) 
    {
        //Traverse from right to left.
        for (size_t i = s.size() - 1; i > 0; i--)
        {
            int bit = (s[i] - '0') + carry;

            if (bit % 2 == 0)
            {
                //Divide by 2
                steps += 1;
            }
            else
            {
                carry = 1;
                steps += 2;

            }
        }

        return steps + carry;
    }

private:
    int steps = 0;
    int carry = 0;
};